src/react/hooks/use-chat-client.ts

```diff
@@ -0,0 +1,29 @@
+import { ChatClient } from '@ably/chat';
+import * as Ably from 'ably';
+import React from 'react';
+
+import { ChatClientContext } from '../contexts/chat-client-context.js';
+import { DEFAULT_CHAT_CLIENT_ID } from '../providers/chat-client-provider.js';
+
+/**
+ * Hook to access the chat client provided by `ChatClientContext`.
+ * This hook must be used within a `ChatClientProvider`.
+ *
+ * @param {string} chatClientId - Identifier of the chat context.
+ * @defaultValue 'default' - This will use the default chat context.
+ *
+ * @throws {ErrorInfo} When the hook is not used within a `ChatClientProvider`.
+ *
+ * @returns {ChatClient} The chat client instance provided by the context.
+ *
+ */
+export const useChatClient = (chatClientId?: string): ChatClient => {
```

Comment:
I think we need symmetry here: if we provide a way to use different chat clients by naming them in the provider, we should also provide a way to fetch them with the hook?

These are edge case uses but I can imagine one might want to
```tsx
<ChatClientProvider client={clientOne} clientId={"client_one"}>
<ChatClientProvider client={clientTwo} clientId={"client_two"}>
{ // rest of the app }
</ChatClientProvider>
</ChatClientProvider>
```

And in the rest of the app randomly use clientOne or clientTwo as they wish.

The alternative is to need to wrap components with `<ChatClientProvider>` everywhere, in which case what's the point of the `clientId` stuff anyway, because you can just get yourself a static class and shove the two clients there yourself.

It's likely I misunderstand react but I'm guessing if you do

```
<ChatClientProvider client={clientOne}>
  { // some components }
    <ChatClientProvider client={clientTwo}>
      { // here }
    </ChatClientProvider>
  { // closing some components }
</ChatClientProvider>
```
then in the `// here` you'll get `clientTwo` and not `clientOne`?

Reply:
Well, after writing the above comment: what's the point of `clientId` anyway, because if the `client` is always required in the provider, `clientId` doesn't really help?

Reply:
Sorry for the many comments. Last thing. If we keep it we should rename `clientId` not to confuse it with auth `clientId`.

Reply:
Removed in both the client provider and the hook, we can add this in again should we ever want to provide support for this.